tests/unit/test.program.js

```diff
@@ -841,6 +841,32 @@ describe('program.main', () => {
     }
   );
 
+  describe('--no-input', () => {
+    const fakeCommands = fake(commands, {
+      run: () => Promise.resolve(),
+    });
+
+    const testCases = [
+      ['--no-input', {noInput: true}],
+      ['--no-input=false', {noInput: false}],
+      ['--no-input=true', {noInput: true}],
+      ['--input', {noInput: false}],
+      ['--input=false', {noInput: true}],
```

Comment:
If we don't support `--input`, then `noInput` should be false here.

Reply:
it is implicitly supported even if hidden, and I'm really not sure that there is an easy way to don't support it without messing with yargs in ways that I would prefer to avoid.

I decided to explicitly included test cases for it because at least we can ensure that it wouldn't regress without being noticed.